cmd/util/cmd/rollback-executed-height/cmd/rollback_executed_height.go

```diff
@@ -230,74 +235,70 @@ func removeForBlockID(
+		return nil, fmt.Errorf("could not find result for block %v: %w", blockID, err)
 	}
 
+	chunkIDs := make([]flow.Identifier, 0, len(result.Chunks))
 	for _, chunk := range result.Chunks {
 		chunkID := chunk.ID()
-		// remove chunk data pack
-		err := chunks.BatchRemove(chunkID, chunkBatch)
-		if err != nil {
-			return fmt.Errorf("could not remove chunk id %v for block id %v: %w", chunkID, blockID, err)
-		}
-
+		chunkIDs = append(chunkIDs, chunkID)
 	}
 
 	// remove commits
-	err = commits.BatchRemoveByBlockID(blockID, writeBatch)
+	err = commits.BatchRemoveByBlockID(blockID, protocolDBBatch)
 	if err != nil {
 		if errors.Is(err, storage.ErrNotFound) {
-			return fmt.Errorf("could not remove by block ID %v: %w", blockID, err)
+			return nil, fmt.Errorf("could not remove by block ID %v: %w", blockID, err)
 		}
 
 		log.Warn().Msgf("statecommitment not found for block %v", blockID)
 	}
 
 	// remove transaction results
-	err = transactionResults.BatchRemoveByBlockID(blockID, writeBatch)
+	err = transactionResults.BatchRemoveByBlockID(blockID, protocolDBBatch)
 	if err != nil {
-		return fmt.Errorf("could not remove transaction results by BlockID %v: %w", blockID, err)
+		return nil, fmt.Errorf("could not remove transaction results by BlockID %v: %w", blockID, err)
 	}
 
 	// remove own execution results index
-	err = myReceipts.BatchRemoveIndexByBlockID(blockID, writeBatch)
+	err = myReceipts.BatchRemoveIndexByBlockID(blockID, protocolDBBatch)
 	if err != nil {
 		if !errors.Is(err, storage.ErrNotFound) {
-			return fmt.Errorf("could not remove own receipt by BlockID %v: %w", blockID, err)
+			return nil, fmt.Errorf("could not remove own receipt by BlockID %v: %w", blockID, err)
 		}
 
 		log.Warn().Msgf("own receipt not found for block %v", blockID)
 	}
 
 	// remove events
-	err = events.BatchRemoveByBlockID(blockID, writeBatch)
+	err = events.BatchRemoveByBlockID(blockID, protocolDBBatch)
 	if err != nil {
-		return fmt.Errorf("could not remove events by BlockID %v: %w", blockID, err)
+		return nil, fmt.Errorf("could not remove events by BlockID %v: %w", blockID, err)
 	}
 
 	// remove service events
-	err = serviceEvents.BatchRemoveByBlockID(blockID, writeBatch)
+	err = serviceEvents.BatchRemoveByBlockID(blockID, protocolDBBatch)
 	if err != nil {
-		return fmt.Errorf("could not remove service events by blockID %v: %w", blockID, err)
+		return nil, fmt.Errorf("could not remove service events by blockID %v: %w", blockID, err)
 	}
 
 	// remove execution result index
-	err = results.BatchRemoveIndexByBlockID(blockID, writeBatch)
+	err = results.BatchRemoveIndexByBlockID(blockID, protocolDBBatch)
 	if err != nil {
 		if !errors.Is(err, storage.ErrNotFound) {
-			return fmt.Errorf("could not remove result by BlockID %v: %w", blockID, err)
+			return nil, fmt.Errorf("could not remove result by BlockID %v: %w", blockID, err)
 		}
 
 		log.Warn().Msgf("result not found for block %v", blockID)
 	}
 
```

Comment:
I would include the chunk data pack index here:
* we first read the IDs of chunk data packs ([this logic](https://github.com/onflow/flow-go/blob/52b05be05d5439239eb6d4386922f669336b7774/storage/store/chunk_data_packs.go#L151-L165))  
* essentially then we remove the index ([these lines](https://github.com/onflow/flow-go/blob/52b05be05d5439239eb6d4386922f669336b7774/storage/store/chunk_data_packs.go#L178-L184))
* In other words, we would be only remove the chunk data pack [index `chunkID -> chunkDataPackID`](https://github.com/onflow/flow-go/blob/52b05be05d5439239eb6d4386922f669336b7774/storage/operation/chunk_data_packs.go#L13-L16) here.
* and we would return the IDs of the chunk data packs encountered.